distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -726,6 +729,85 @@ type Table
+         of term in each individual table cell. If set to `All`, it replaces all
+         instances of term.
+       - matcher: If a `Text_Matcher`, the text is compared using case-sensitivity
+         rules specified in the matcher. If a `Regex_Matcher`, the term is used as a
+         regular expression and matched using the associated options.
+       - on_problems: Specifies how to handle if a problem occurs, raising as a
+         warning by default.
+
+         The following problems can occur:
+         - If a column in columns is not in the input table, a `Missing_Input_Columns`.
+         - If duplicate columns, names or indices are provided, a
+           `Duplicate_Column_Selectors`.
+         - If a column index is out of range, a `Column_Indexes_Out_Of_Range`.
+         - If two distinct indices refer to the same column, an
+           `Input_Indices_Already_Matched`, with the column included the first
+           time it is matched.
+         - If a column in columns does not have a storage type of `Text`, or `Any`,
+           thus it is guaranteed that it can't contain any text values, a
+           `Invalid_Value_Type`.
+
+       > Example
+         Replace dashes with underscores in a column named "variable_names".
+
+             table.replace_text "variable_names" "-" "_" 
+
+       > Example
+         Remove leading and trailing spaces from cells in multiple columns.
+
+             table.replace_text (Column_Selector.By_Name ["foo", "bar"]) "^\s*(.*?)\s*$" "$1" matcher=Regex_Matcher.Regex_Matcher_Data
+
+       > Example
+         Replace texts in quotes with parentheses in column at index 1.
+
+             table.replace_text 1 '"(.*?)"' '($1)' matcher=Regex_Matcher.Regex_Matcher_Data
+    replace_text : (Text | Integer | Column_Selector) -> Text -> Text -> Matching_Mode | Regex_Mode -> (Text_Matcher | Regex_Matcher) -> Problem_Behavior -> Table
+    replace_text self columns=(Column_Selector.By_Index [0]) term="" new_text="" mode=Regex_Mode.All matcher=Text_Matcher.Case_Sensitive on_problems=Problem_Behavior.Report_Warning = if term.is_empty then self else
+        problem_builder = Problem_Builder.new
+        
+        selector = case columns of
+            _ : Column_Selector -> columns
+            name : Text -> Column_Selector.By_Name [name]
+            index : Integer -> Column_Selector.By_Index [index]
+        selection = Table_Helpers.select_columns_helper self.columns selector reorder=False problem_builder
+        selected_names = Map.from_vector (selection.map column-> [column.name, True])
+
+        map_preserve_name column f = column.map f . rename column.name
+        do_replace = _.replace term new_text mode matcher
+        do_replace_only_text = case _ of
+                item : Text -> do_replace item
+                item -> item
+
+        transform column = case column.storage_type of
+            Storage.Text -> map_preserve_name column do_replace
```

Comment:
I'd think that for the column backed by `StringStorage` it may be worth implementing a Java counterpart so that the iteration can happen in pure Java and be significantly faster. `Column.map` is unfortunately not very efficient as it is Java code calling into an Enso callback back-and-forth and that incurs a significant "context-switch" cost.

However, I'm not sure if this is in scope of this PR - so as long as this works OK, I think maybe we can leave it as-is and create a ticket to consider improving it sometime in the future. The downside is that here we rely quite a lot on the `Text.replace` logic which would have to be replicated or moved to Java to be callable there.

Reply:
Agree we will want to improve this in future work (part of adding `Column.text_replace` work). Lets leave it for another PR.